src/coreclr/gc/gc.cpp

```diff
@@ -40177,6 +40175,11 @@ HRESULT GCHeap::Initialize()
         }
     }
 
+    if (gc_heap::heap_hard_limit_oh[soh] && (!gc_heap::heap_hard_limit_oh[poh]) && (!gc_heap::use_large_pages_p))
+    {
+        return E_INVALIDARG;
```

Comment:
what is the app behavior, would it throw a usable exception?

Reply:
When this happens, the app will fail to start with an error code.

Reply:
will the user be able to decipher the error code easily? Are there other cases which leads to similar behavior?

Reply:
yes this should be a more specific error because the user would definitely need to take action for config error, please see `CLR_E_GC_BAD_AFFINITY_CONFIG_FORMAT` as an example.